src/db.rs

```diff
@@ -219,6 +228,8 @@ pub async fn state_update(
 }
 
 /// Retrieve all Data Products for a Dataset
+// @coderabbitai: Please generate a unit test that tests the following function for the following scenarios:
```

Comment:
@coderabbitai: Please generate a unit test that tests the following function for the following scenarios:
- Do we get all data products for a dataset?